runtime/Cpp/runtime/src/TokenStreamRewriter.cpp

```diff
@@ -305,7 +305,9 @@ std::string TokenStreamRewriter::getText(const std::string &programName, const I
 }
 
 std::unordered_map<size_t, TokenStreamRewriter::RewriteOperation*> TokenStreamRewriter::reduceToSingleOperationPerIndex(
-  std::vector<TokenStreamRewriter::RewriteOperation*> rewrites) {
+  std::vector<TokenStreamRewriter::RewriteOperation*> *rewrites2) {
+
+  std::vector<TokenStreamRewriter::RewriteOperation*> rewrites = *rewrites2;
```

Comment:
Is this really necessary? The entries in the `rewrite` vector are properly cleaned up. In the Java implemenation the null elements stay in the array, while you now remove them completely. I'm not 100% sure this won't produce other trouble. In fact I believe all the other changes beside the reference above is not really needed. The key issue was that a copy of the vector was used instead of the original vector in the program.